Source/DafnyCore/AST/Types/ArrowType.cs

```diff
@@ -43,6 +41,21 @@ public class ArrowType : UserDefinedType {
     }
   }
 
+  public static void ComputeResolvedOpcodeAndBoundedPool(Expression collection, CollectionType collectionType,
+    out BinaryExpr.ResolvedOpcode resolvedOpcode, out ComprehensionExpr.BoundedPool boundedPool) {
+    if (collectionType is SetType setType) {
+      resolvedOpcode = BinaryExpr.ResolvedOpcode.InSet;
```

Comment:
Maybe this opCode can be tied to the BoundedPool ? `CollectionBoundedPool.ResolvedOpCode`.

Reply:
In general, there may be other bounded pools associated with an `in` expression, too. So, I just kept them together like they were.

Reply:
Sure, but you can still let the `BoundedPool` say which opcode should be used with it right?